### ISSUE TITLE
[Bug]: Do something immediately to address no packages published for 5.71.0

### ISSUE THREAD

**Issue by et304383** [hashicorp/terraform-provider-aws#39724]
### Terraform Core Version

1.6.6

### AWS Provider Version

5.71.0

### Expected Behavior

I pull SOMETHING or I pull 5.72.0 TODAY.

### Actual Behavior

Error saying no providers for this version.

### Relevant Error/Panic Output Snippet

```shell
Initializing the backend...

Initializing provider plugins...
- Finding hashicorp/aws versions matching "~> 5.0"...
╷
│ Error: Incompatible provider version
│ 
│ Provider registry.terraform.io/hashicorp/aws v5.71.0 does not have a package available for your current platform, darwin_arm64.
│ 
│ Provider releases are separate from Terraform CLI releases, so not all providers are available for all platforms. Other versions of this provider may have different platforms supported.
╵
```

### Terraform Configuration Files

N/A

### Steps to Reproduce

Pin provider to `~> 5.0` and run terraform init

### Would you like to implement a fix?

None

**Comment by et304383** [comment]
Waiting until Thursday when you publish a new version isn't acceptable.  We cannot have builds blocked for two days and we cannot feasibly go and update hundreds of versions.tf files to pin to 5.70.0

**Comment by jasonm3-netapp** [comment]
Hashicorp could you please check to see if you pulled more than intended? We're pinned to 5.65.0 and we're getting no available version meeting constraints.

```
│ Error: Failed to query available provider packages
│ 
│ Could not retrieve the list of available versions for provider
│ hashicorp/aws: no available releases match the given constraints >= 2.23.0,
│ >= 3.27.0, >= 3.29.0, >= 3.70.0, >= 3.72.0, >= 4.0.0, >= 4.9.0, >= 4.19.0,
│ >= 4.21.0, >= 4.22.0, >= 4.33.0, >= 4.36.0, >= 4.40.0, >= 4.54.0, >=
│ 4.57.0, >= 4.59.0, >= 4.63.0, >= 4.64.0, >= 4.66.0, >= 5.0.0, >= 5.9.0, >=
│ 5.27.0, >= 5.61.0, >= 5.65.0, 5.65.0, >= 5.70.0
```

**Comment by stevehipwell** [comment]
I'm sorry, but who thinks that leaving the tag and deleting the artifacts is going to be a solution? By pulling the release you've broken Dependabot updates as the tag still exists. This is the worst possible fix as all of us who were working correctly with this release are all broken and left to manually resolve this.

Deleting the tag might allow a rollback but the safest course of action would be to do a patch release for `5.71.1` on the `5.70.0` commit which would allow automation to upgrade forwards.

**Comment by lorengordon** [comment]
Also just ran into this. I updated the lock file on my projects to 5.71.0 just yesterday 😭

**Comment by et304383** [comment]
@justinretzolk

**Comment by atheiman** [comment]
Yep we ran into `Error: Resource instance managed by newer provider version` also after auto-upgrading to 5.71 when it was released. Now our tf projects are broken b/c `aws_backup_plan` added a new attribute in 5.71 provider, so `aws_backup_plan` resources cannot be read by 5.70 provider.

**Comment by bryanmorganoverbey** [comment]
Same thing for us as others have said above. 
`│ Error: Resource instance managed by newer provider version
│ 
│ The current state of
│ module.backup_vault_artifacts_v2.aws_backup_plan.daily_backup was created
│ by a newer provider version than is currently selected. Upgrade the aws
│ provider to work with this state.`

A v5.71.1 release would fix our issue.

**Comment by jamengual** [comment]
Time to switch to opentofu

**Comment by Wolfsrudel** [comment]
> Time to switch to opentofu

Didn't know they had their own AWS provider. Pointless comment.

**Comment by justinretzolk** [comment]
Hey everyone :wave: Thank you for continuing to update us on the behavior you've experienced -- that sort of information is always extremely useful to us. I've been monitoring this issue for the past few hours while we continued discussions internally, and I'd like to provide some updates.

I'll lead with the most important part: we're in the process of releasing `5.72.0` now. This release will include all of the changes from the now-removed `5.71.0`, as well as a few additional items that have been completed in the meantime. For more specifics, refer to the [CHANGELOG](https://github.com/hashicorp/terraform-provider-aws/blob/main/CHANGELOG.md).

Given the impact, I also want to provide as much information as I can about what occurred here. On Thursday (October 10th), we attempted to release `5.71.0`. The release encountered errors, so additional attempts were made to cut the release. Overnight, several reports came in detailing checksum errors for various platforms. To that point, it seemed that downloading and then actually using `5.71.0` was impossible, so we made the decision to pull the release from the Registry. Knowing that we had not yet found the root cause of the issue, that we were coming up on a long weekend, and that HashiConf was this week, we made the decision not to release on Friday, lest we make matters worse.

Over the weekend, the Terraform Registry detected the `5.71.0` release on GitHub and appears to have re-ingested the provider version. Initial reports of continued issues appeared -- initially -- to be caching issues, and so I treated them as such, responding and closing them. This morning, after this issue was raised, we continued our mitigation efforts. Namely, we renamed the release so that the Registry wouldn't attempt to pull it in again, and worked with the appropriate teams to ensure all appropriate caches had been cleared. Doing this successfully removed `5.71.0` once and for all, but had the unfortunate effects seen here of blocking anyone who had successfully used `5.71.0` in the periods where it was available. With that in mind, we determined that the most appropriate course of action was to release `5.72.0` as soon as possible, which brings us to where we are today.

I'm going to leave this issue open for now, until we hear back that the issues tied to this have been resolved, and that those who have been blocked are unblocked. In the meantime, I hope that the information I've laid out helps give a better idea of where things are at the moment.

**Comment by et304383** [comment]
@justinretzolk why can't you push 5.70.0 as 5.71.1 to get around all this until you can push 5.72.0?

**Comment by justinretzolk** [comment]
@et304383 -- `5.72.0` was in flight by the time I left my comment (and has now been released). Pushing the code from `5.70.0` as `5.71.1` would have been a quick fix, but wouldn't have resolved all of the issues that had arisen out of this. For example #39730, where `5.71.0` introduced changes to `aws_backup_plan` that would have been incompatible with the newer `5.71.1` version.

I can see where you're coming from, but it's also important to us to ensure we didn't make a bad situation worse as a result of trying to get something out as quickly as possible.

**Comment by bryanmorganoverbey** [comment]
@justinretzolk Our issue is solved with the new version and our ci-cd pipeline is deploying again. Thank you so much for the expediency!

**Comment by asvinours** [comment]
Looking good on our side too, terraform is downloading `5.72.0` without issues. Thank you :pray:

**Comment by atheiman** [comment]
I am now pulling hashicorp/aws v5.72.0 after a terraform init -upgrade and am no longer seeing issues related to `aws_backup_plan` new attribute. Thanks for the quick fix.

**Comment by stevehipwell** [comment]
@justinretzolk could you please explain what testing you did to guide your course of action? Based on my observations across a large number of TF workspaces your intervention caused significantly more negative impact than the original issue.

**Comment by et304383** [comment]
@justinretzolk I am glad this is now resolved with 5.72.0 but in the future Hashicorp should have prioritized fixing this immediately above all other issues even if it meant pushing up 5.70.0 as 5.71.1.  This is a case where pushing through to get the next release out caused hours (days?) of failed builds across the world for numerous companies, most of who probably did not find this issue or even have an understanding of what was going on.

I know that your all work hard to provide this software free of charge, but at the same time we've all come to depend on certain aspects just working, including download of the latest version.
